src/amo/components/SearchPage.js

```diff
@@ -17,17 +20,28 @@ export default class SearchPage extends React.Component {
   }
 
   render() {
-    const { count, lang, loading, page, query, results } = this.props;
-    const pathname = `/${lang}/firefox/search/`;
+    const {
+      clientApp, count, lang, loading, page, query, results,
+    } = this.props;
+    const pathname = `/${lang}/${clientApp}/search/`;
```

Comment:
I think if you make `Paginate` accept a `Link` prop or use our internal `Link` component you could just pass it `/search/` and remove the redux stuff from this component.